src/client/client.ts

```diff
@@ -178,6 +178,25 @@ export default class TaskClient {
 
         const now = Date.now();
 
+        // Calculate nextRunTime
+        let nextRunTime = computeNextRun(
+          definedOptions.interval,
+          undefined,
+          definedOptions.taskEndTime
+        );
+        if (definedOptions.scheduledTime) {
+          if (definedOptions.taskEndTime) {
+            if (
+              definedOptions.taskEndTime.getTime() <=
+              definedOptions.scheduledTime.getTime()
+            ) {
+              nextRunTime = undefined;
+            }
+          } else {
+            nextRunTime = definedOptions.scheduledTime.getTime();
+          }
```

Comment:
isn't this equivalent to calling `isValidNextRun(definedOptions.scheduledTime, definedOptions.taskEndTime)`? might make more sense to just export that function and use it here